src/main/java/de/tum/in/www1/artemis/service/EmailSummaryService.java

```diff
@@ -0,0 +1,114 @@
+package de.tum.in.www1.artemis.service;
+
+import static de.tum.in.www1.artemis.service.notifications.NotificationSettingsService.NOTIFICATION__WEEKLY_SUMMARY__BASIC_WEEKLY_SUMMARY;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.repository.NotificationSettingRepository;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+
+@Service
+public class EmailSummaryService {
+
+    private final MailService mailService;
+
+    private final NotificationSettingRepository notificationSettingRepository;
+
+    private final CourseService courseService;
+
+    private final ExerciseRepository exerciseRepository;
+
+    private Duration scheduleInterval;
+
+    private final Duration weekly = Duration.ofDays(7);
+
+    public EmailSummaryService(MailService mailService, NotificationSettingRepository notificationSettingRepository, CourseService courseService,
+            ExerciseRepository exerciseRepository) {
+        this.mailService = mailService;
+        this.notificationSettingRepository = notificationSettingRepository;
+        this.courseService = courseService;
+        this.exerciseRepository = exerciseRepository;
+    }
+
+    /**
+     * Begin the process of email summaries
+     * i.e. find all active Artemis users that have weekly summaries enabled in their notification settings
+     * and initiate the creation of summary emails for each found user.
+     */
+    @Async
+    public void prepareEmailSummaries() {
+        checkSecurityUtils();
```

Comment:
Is there a reason for calling `checkSecurityUtils()` method both here and at [`scheduleEmailSummaries()`](https://github.com/ls1intum/Artemis/pull/4512/files#diff-19fc1f340b973371be268212ba65f841d380a5d91015a1585fbaef58887b2cc4R83)? Are there different contexts involved?

Reply:
Good question,
I use this here because it is a method that is called asynchronously and from my experience, it will not work otherwise.
In the runnable function in scheduleEmailSummaries I simply followed other Runnables in our code stack : e.g. 
![image](https://user-images.githubusercontent.com/65814168/157914851-2fc82d40-92e8-4e24-b4f2-59fa0f0dd4ea.png)
![image](https://user-images.githubusercontent.com/65814168/157914855-801dac1b-2b08-4784-bddd-11893ae9dc48.png)
Because this runnable is the function that is called by the scheduler we need to get the authentication again (at least from my experience so far working with asyn and schedulers)

Reply:
Thank you for the detailed explanation, in that case everything looks good as far as I can see.